### PR TITLE
Support arbitary depth config overloading of DOM properties

### PR DESCRIPTION
**Pull request by jonathanKingston**
See: https://app.asana.com/0/0/1204271448367771/f

Allows us to override properties to any depth within the DOM.

### CODE REVIEW

File: src/features/runtime-checks.js [hunk 1]
```diff
@@ -157,32 +162,64 @@ class DDGRuntimeChecks extends HTMLElement {
...
+         */
+        function stringifyScope (scope, scopePath) {
+            let output = ''
+            for (const [key, value] of scope) {
+                const varOutName = [...scopePath, key].join('_')
+                if (value instanceof Map) {
+                    const proxyName = `_proxyFor_${varOutName}`
+                    output += `
+                    let ${proxyName}
+                    if (${scopePath.join('?.')}?.${key} === undefined) {
```
Comment (shakyShane) [inline src/features/runtime-checks.js:193]: with this not be transpiled, I'm assuming you're ok with the support level of optional chaining?

**Review by shakyShane — DISMISSED** [review]
My only comments on this would be around testing the string output in the future too, to compliment the integration testing. 
 
Since we're just composing a complicated string - I think it would be good to have a base output to compare against (like an actual JS file). It might help surface any problems and would be good for understanding the intention.

On the other hand, it's very hard to construct a string that can be compared against a base one - let's just keep it in mind imo

File: src/features/runtime-checks.js (hunk 1 above)
Comment (jonathanKingston) [reply src/features/runtime-checks.js:193]: 👀 checks notes

Comment (jonathanKingston) [reply src/features/runtime-checks.js]: I think we're good based on the pixels I'm seeing.

**Review by shakyShane — APPROVED** [review]
Nice!

**Pull request merged**